pkg/target/manager.go

```diff
@@ -0,0 +1,399 @@
+}
+
+func (m *managerImpl) TargetProject(ctx context.Context, projectName string) error {
+	return m.patchTarget(func(t *targetImpl) error {
+		if t.Garden == "" {
+			return ErrNoGardenTargeted
+		}
+
+		gardenClient, err := m.clientForGarden(t.Garden)
+		if err != nil {
+			return fmt.Errorf("could not create Kubernetes client for garden cluster: %w", err)
+		}
+
+		// validate that the project exists
+		if _, err := m.resolveProjectName(ctx, gardenClient, projectName); err != nil {
+			return fmt.Errorf("failed to validate project: %w", err)
+		}
+
+		t.Seed = ""
+		t.Project = projectName
+		t.Shoot = ""
+
+		return nil
+	})
+}
+
+func (m *managerImpl) resolveProjectName(ctx context.Context, gardenClient client.Client, projectName string) (*gardencorev1beta1.Project, error) {
+	// validate that the project exists
+	project := &gardencorev1beta1.Project{}
+	key := types.NamespacedName{Name: projectName}
+	err := gardenClient.Get(ctx, key, project)
+
+	return project, err
+}
+
+func (m *managerImpl) TargetSeed(ctx context.Context, seedName string) error {
+	return m.patchTarget(func(t *targetImpl) error {
+		if t.Garden == "" {
+			return ErrNoGardenTargeted
+		}
+
+		gardenClient, err := m.clientForGarden(t.Garden)
+		if err != nil {
+			return fmt.Errorf("could not create Kubernetes client for garden cluster: %w", err)
+		}
+
+		// validate that the seed exists
+		seed, err := m.resolveSeedName(ctx, gardenClient, seedName)
+		if err != nil {
+			return fmt.Errorf("failed to validate seed: %w", err)
+		}
+
+		t.Seed = seedName
+		t.Project = ""
+		t.Shoot = ""
+
+		if err := m.updateSeedKubeconfig(ctx, gardenClient, t, seed); err != nil {
+			return fmt.Errorf("failed to fetch kubeconfig for seed cluster: %w", err)
+		}
+
+		return nil
+	})
+}
+
+func (m *managerImpl) updateSeedKubeconfig(ctx context.Context, gardenClient client.Client, t Target, seed *gardencorev1beta1.Seed) error {
+	// fetch kubeconfig secret
+	secret := corev1.Secret{}
+	key := types.NamespacedName{Name: seed.Spec.SecretRef.Name, Namespace: seed.Spec.SecretRef.Namespace}
+
+	if err := gardenClient.Get(ctx, key, &secret); err != nil {
+		return fmt.Errorf("failed to retrieve seed kubeconfig: %w", err)
+	}
+
+	return m.kubeconfigCache.Write(t, secret.Data["kubeconfig"])
+}
+
+func (m *managerImpl) resolveSeedName(ctx context.Context, gardenClient client.Client, seedName string) (*gardencorev1beta1.Seed, error) {
+	seed := &gardencorev1beta1.Seed{}
+	key := types.NamespacedName{Name: seedName}
+
+	if err := gardenClient.Get(ctx, key, seed); err != nil {
+		return nil, err
+	}
+
+	if seed.Spec.SecretRef == nil {
+		return nil, errors.New("spec.SecretRef is missing in this seed, seed not reachable")
+	}
+
+	return seed, nil
+}
+
+func (m *managerImpl) TargetShoot(ctx context.Context, shootName string) error {
+	return m.patchTarget(func(t *targetImpl) error {
+		if t.Garden == "" {
+			return ErrNoGardenTargeted
+		}
+
+		if t.Project == "" && t.Seed == "" {
```

Comment:
as written here https://github.com/gardener/gardenctl-v2/pull/5/files#r640661421 it shouldn't be a hard requirement that a project or seed is targeted. We could figure the out the project when searching for the shoot across all namespaces